### PR TITLE
Allow gblocs and markets to be null in the DB

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

GBLOCs and Markets on the Shipment model are allowed to be null.  This updates the payload for the public and internal shipment api's to allow for that and to avoid null dereferencing like we're seeing in staging.

## Reviewer Notes

Are there other models where this is an issue?

## Setup

First populate the DB:

```sh
make db_dev_e2e_populate
```

Log into the TSP app and find a shipment you have access to.  With the shipment ID in hand modify the DB:

```
./bin/psql-dev
```

```sql
update shipments set destination_gbloc=NULL where id='76e4e692-fb7d-46a2-ab00-16482dcc6f1a';
```

Then try to get the shipment from the API: http://tsplocal:3000/api/v1/docs#/shipments/getShipment

## Code Review Verification Steps

* [x] Request review from a member of a different team.

## References

* [Pivotal story](tbd) for this change

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
I feel like maybe we need a test that creates a shipment record with every field that can possibly be nullable set to `NULL` and then we test out the handler.  WDYT?

File: pkg/handlers/internalapi/shipments_test.go [hunk 1]
```diff
@@ -14,6 +14,71 @@ import (
...
+	shipment.PmSurveyCompletedAt = nil
+	shipment.PmSurveyPlannedPackDate = nil
+	shipment.PmSurveyPlannedPickupDate = nil
+	shipment.PmSurveyPlannedDeliveryDate = nil
+	shipment.PmSurveyWeightEstimate = nil
+	shipment.PmSurveyProgearWeightEstimate = nil
+	shipment.PmSurveySpouseProgearWeightEstimate = nil
+	shipment.PmSurveyNotes = nil
+
+	shipmentPayload, _ := payloadForShipmentModel(shipment)
```
Comment (mikena-truss) [inline pkg/handlers/internalapi/shipments_test.go]: Since `pyarloadForShipmentModel` is private, I'm wondering if what we really want to test is the handler response. Would that be difficult to do?

File: swagger/api.yaml [hunk 2]
```diff
@@ -973,7 +974,6 @@ definitions:
       market:
         $ref: '#/definitions/ShipmentMarket'
         readOnly: true
-        x-nullable: true
```
Comment (mikena-truss) [inline swagger/api.yaml]: Why is this one getting removed?

File: pkg/handlers/internalapi/shipments_test.go (hunk 1 above)
Comment (chrisgilmerproj) [reply pkg/handlers/internalapi/shipments_test.go]: I had considered testing the handler response but really the problem is scoped to this private method.  I don't really mind having a test for private methods and I think it will be more clear what the problem is if we test these (and hopefully others like it) directly.

File: swagger/api.yaml (hunk 2 above)
Comment (chrisgilmerproj) [reply swagger/api.yaml]: The definition it refers to has the `x-nullable` property, so we don't need to define it twice.
